test-data/src/test/resources/org/openmrs/module/fhir2/api/dao/impl/FhirEncounterDaoImplTest_initial_data.xml

```diff
@@ -8,55 +8,26 @@
     Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
     graphic logo is a trademark of OpenMRS Inc.
 -->
-<dataset>
```

Comment:
So, in OpenMRS we have a bad habit (me included, in the past) of muddling custom test data added for a specific test with the standard test data set.  A lot of this is likely due to the fact that people don't realize the standard test dataset is available to them.

However, if you define an element in some custom data with the same primary key as an element the standard test data, it overrides that test data element, making it particularly difficult to determine what data is coming from where.

it's particularly problematic when testing search elements and looking through  both the standard test data and any custom data to determine exactly what elements should be in the result set.

To this end, I went down a bit of a rabbit hole and stripped out *all* the test data from this file except for the new data added specifically to test the new encounterWithMedicationRequests.    Then I tweaks all the existing tests so that they properly tested using only the standard test data set, which turned out to be a bit more of a rabbit hole than I though, but i think I got it sorted!

Reply:
Hopefully it turns out to be an actual improvement!